frontend/lib/parsing/parsing-queries.cpp

```diff
@@ -388,32 +388,71 @@ void countTokens(Context* context, UniqueString path, ParserStats* parseStats) {
   }
 }
 
-const Location& locateId(Context* context, ID id) {
-  QUERY_BEGIN(locateId, context, id);
-
-  Location result;
+static const BuilderResult*
+builderResultOrNull(Context* context, ID id, UniqueString& pathOut) {
+  UniqueString parentSymbolPath;
 
   // Ask the context for the filename from the ID
-  UniqueString path;
-  UniqueString parentSymbolPath;
+  bool found = context->filePathForId(id, pathOut, parentSymbolPath);
 
-  bool found = context->filePathForId(id, path, parentSymbolPath);
   if (found) {
     // Get the result of parsing
-    const BuilderResult& p = parseFileToBuilderResult(context, path,
-                                                      parentSymbolPath);
-    result = p.idToLocation(id, path);
+    const BuilderResult& br = parseFileToBuilderResult(context, pathOut,
+                                                       parentSymbolPath);
+    return &br;
+  }
+
+  return nullptr;
+}
+
+const Location& locateId(Context* context, ID id) {
+  QUERY_BEGIN(locateId, context, id);
+  Location result;
+
+  UniqueString path;
+  if (auto br = builderResultOrNull(context, id, path)) {
+    result = br->idToLocation(id, path);
   }
 
   return QUERY_END(result);
 }
 
 // this is just a convenient wrapper around locating with the id
 const Location& locateAst(Context* context, const AstNode* ast) {
+  CHPL_ASSERT(ast);
   CHPL_ASSERT(!ast->isComment() && "cant locate comment like this");
+  // TODO: This will crash if 'ast' is null, can we fix that?
   return locateId(context, ast->id());
```

Comment:
IMO, `null` is just not a valid input to `locateAst`, so there's nothing to be done. Garbge in, garbage out.